pallets/msa/src/lib.rs

```diff
@@ -1097,6 +1097,11 @@ impl<T: Config> Pallet<T> {
 		payload: Vec<u8>,
 	) -> DispatchResult {
 		let key = T::ConvertIntoAccountId32::convert((*signer).clone());
+
+		if signature.verify(&payload[..], &key) {
+			return Ok(())
+		}
+
 		let wrapped_payload = wrap_binary_data(payload);
```

Comment:
Hmm... Not sure how heavy the verification step is, but do we want to assume wrapped or unwrapped as more common? (aka which one to try first?)

Reply:
Good point! I’m thinking the optimization might not matter much here since we base our weights on the worst-case scenario. This flow seems like it would be the worst case, so it might make more sense to focus on handling that rather than optimizing for the less common cases.